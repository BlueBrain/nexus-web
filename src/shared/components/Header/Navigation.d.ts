import * as React from 'react';
import './Navigation.less';
declare const Navigation: React.FC<{
  subApps: any;
  authenticated: boolean;
}>;
export default Navigation;
