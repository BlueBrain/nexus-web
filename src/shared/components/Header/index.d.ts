import * as React from 'react';
import { Realm } from '@bbp/nexus-sdk';
import { ConsentType } from '../../layouts/FusionMainLayout';
import './Header.less';
export interface HeaderProps {
  version: string;
  githubIssueURL: string;
  forgeLink: string;
  name?: string;
  token?: string;
  links?: React.ReactNode[];
  realms: Realm[];
  serviceAccountsRealm: string;
  displayLogin?: boolean;
  children?: React.ReactChild;
  consent?: ConsentType;
  commitHash?: string;
  dataCart?: React.ReactNode;
  onClickRemoveConsent?(): void;
  performLogin(realmName: string): void;
  subApps: any;
  authenticated: boolean;
}
declare const Header: React.FunctionComponent<HeaderProps>;
export default Header;
