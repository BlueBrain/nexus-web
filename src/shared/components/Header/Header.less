@import '../../lib.less';

.Header {
  display: flex;
  justify-content: space-between;
  box-sizing: content-box;
  height: 52px;
  background-color: @fusion-menu-color;
  position: fixed;
  width: 100%;
  z-index: 20;
  padding-left: 350px;

  .selectors {
    display: flex;
    justify-content: flex-start;
    align-items: center;
    flex-grow: 1;
    margin: 0 4px;

    .OrgDropdown {
      height: 100%;
    }
  }

  .menu-block {
    display: flex;
    justify-content: flex-end;
    padding-right: @default-pad;
    align-items: center;

    button,
    a {
      margin: 0 4px;
      font-weight: lighter;
    }

    .menu-dropdown {
      align-self: center;
      margin: 5px 5px 4px;
      bottom: 50px;
      color: @fusion-background-color;
      border: 1px solid @fusion-background-color;
      border-radius: @border-radius-base;
      padding: 3px 5px;
      height: 32px;
    }
  }

  .ui-header-info-button {
    margin: 5px 5px 4px;
    cursor: pointer;
    border: 1px solid @fusion-background-color;
    border-radius: @border-radius-base;
    padding: 5px;
    height: 32px;
  }

  .cart {
    color: @fusion-background-color;
    background-color: @fusion-menu-color;
  }

  .copy-token-button {
    height: 32px;
    width: 115px;
    border-radius: @border-radius-base;
    outline-style: none;
    line-height: 1rem;
    cursor: pointe r;
    font-size: 0.9rem;
    text-align: left;
    border: 1px solid @fusion-background-color;
    display: flex;
    align-items: center;
    background-color: inherit;

    .button-text {
      margin-left: 0.4rem;
      color: @fusion-background-color;
    }
  }

  .forge-button {
    height: 32px;
    padding: 0 7px;
    border-radius: @border-radius-base;
    outline-style: none;
    line-height: 1rem;
    cursor: pointer;
    font-size: 0.9rem;
    display: flex;
    align-items: center;
    background-color: @alternative-button-color;
    margin-left: 0.4rem;
    color: #fff;
  }
}

.ant-popover-inner {
  background-color: @fusion-background-color !important;
}
.ant-popover-inner-content {
  padding: 6px !important;
}

.navigation {
  width: 228px;
  .app-icon {
    display: inline-block;
    width: 70px;
    height: 70px;
    text-align: center;
    margin: 3px;
    filter: drop-shadow(0px 0px 1px #ddd);
    .ant-btn {
      padding: 0;
      height: 100%;
      width: 100%;
      border-radius: 4px;
      background-color: @fusion-background-color;
    }
  }
}

.nav-item {
  font-size: 16px;
}

.popover-title {
  padding: 0 0 5px 0;
}

// the gtm modal should appear on a top of a popover
.ant-popover {
  z-index: 1000 !important;
}

/*
  ##Device = Tablets, Ipads (portrait)
  ##Screen = B/w 768px to 1024px
*/

@media (max-width: 1024px) {
  .hide-tablet {
    display: none;
  }
}
