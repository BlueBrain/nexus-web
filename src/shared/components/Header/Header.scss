@import '../../lib.scss';

.Header {
  display: flex;
  justify-content: space-between;
  box-sizing: content-box;
  height: 52px;
  background-color: $fusion-menu-color;
  position: fixed;
  width: 100%;
  z-index: 1000;

  .selectors {
    display: flex;
    justify-content: flex-start;
    align-items: center;
    flex-grow: 1;
    margin: 0 4px;

    .OrgDropdown {
      height: 100%;
    }
  }

  .menu-block {
    display: flex;
    justify-content: flex-end;
    // padding-right: $default-pad;
    align-items: center;
    min-width: 120px;
    button,
    a {
      font-weight: 400;
      font-size: 16px;
    }

    .menu-dropdown {
      display: flex;
      align-items: center;
      justify-content: center;
      align-self: center;
      width: 100%;
      color: $fusion-background-color;
      border: none;
      height: 100%;
      // border-radius: $border-radius-base;
      border-radius: 0 !important;
      padding: 5px 20px;
      // height: 32px;
      background-color: $fusion-main-color;

      span {
        user-select: none;
        margin-left: 5px;
        text-transform: capitalize;
      }
    }
  }

  .menu-open-creation-panel {
    cursor: pointer;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    padding: 0px;
    background: #003a8c;
    height: 100%;
    min-width: 50px;
    width: 100%;

    &:hover {
      svg {
        transform: scale(1.2);
        transition: transform 0.2 ease-out;
      }
    }
  }

  .ui-header-info-button {
    margin: 5px 5px 4px;
    cursor: pointer;
    border: 1px solid $fusion-background-color;
    border-radius: $border-radius-base;
    padding: 5px;
    height: 32px;
  }

  .cart {
    color: $fusion-background-color;
    background-color: $fusion-menu-color;
  }

  .copy-token-button {
    height: 32px;
    width: 115px;
    border-radius: $border-radius-base;
    outline-style: none;
    line-height: 1rem;
    cursor: pointe r;
    font-size: 0.9rem;
    text-align: left;
    border: 1px solid $fusion-background-color;
    display: flex;
    align-items: center;
    background-color: inherit;
    cursor: pointer;

    .button-text {
      margin-left: 0.4rem;
      color: $fusion-background-color;
    }
  }

  .forge-button {
    height: 32px;
    padding: 0 7px;
    border-radius: $border-radius-base;
    outline-style: none;
    line-height: 1rem;
    cursor: pointer;
    font-size: 0.9rem;
    display: flex;
    align-items: center;
    background-color: $alternative-button-color;
    margin-left: 0.4rem;
    color: #fff;
  }

  .logo-container {
    z-index: 21;

    &__logo {
      padding: 0.85rem;
      padding-left: 0.95rem;
      display: flex;
      letter-spacing: 1px;
      color: #ffffff;
      height: 100%;
    }
  }
}

.menu-overlay-custom {
  top: 52px !important;

  .ant-dropdown-menu {
    background-color: $fusion-main-color !important;
    border-top-right-radius: 0;
    border-top-left-radius: 0;
    box-shadow: none !important;
    padding: 0 !important;

    .ant-dropdown-menu-item {
      line-height: 30px !important;
    }

    .ant-dropdown-menu-submenu-title,
    .ant-dropdown-menu-item {
      color: $fusion-daybreak-3;
    }
  }

  .menu-item-logout {
    background-color: $fusion-daybreak-10;
  }
}

.submenu-overlay-custom {
  span {
    line-height: 30px !important;
  }

  &:hover {
    .ant-dropdown-menu-submenu-title {
      background-color: $fusion-daybreak-3 !important;
      color: $fusion-daybreak-9 !important;
    }
  }

  .ant-dropdown-menu-item {
    line-height: 30px !important;
  }

  .ant-dropdown-menu-submenu-expand-icon {
    display: none !important;
  }
}

.submenu-overlay-custom-popup {
  border-radius: 0 !important;

  .ant-dropdown-menu {
    background-color: $fusion-daybreak-3 !important;
    box-shadow: none !important;
    margin-left: 0px !important;
    margin-right: 0px !important;
    padding: 0px 0px;

    .ant-dropdown-menu-item {
      line-height: 30px !important;
      color: $fusion-daybreak-9 !important;
    }
  }
}

.link-menu-item {
  a {
    color: $fusion-daybreak-3 !important;
  }
}

.ant-popover-inner {
  background-color: $fusion-background-color !important;
}

.ant-popover-inner-content {
  padding: 6px !important;
}

.popover-title {
  padding: 0 0 5px 0;
}

.ant-popover {
  z-index: 1000 !important;
}

@media (max-width: 1024px) {
  .hide-tablet {
    display: none;
  }
}
