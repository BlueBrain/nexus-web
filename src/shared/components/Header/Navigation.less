@import '../../lib.less';

.navigation {
  width: 228px;
  &__item {
    display: inline-block;
    width: 70px;
    height: 70px;
    text-align: center;
    margin: 3px;
    filter: drop-shadow(0px 0px 1px #ddd);
    .ant-btn {
      padding: 0;
      height: 100%;
      width: 100%;
      border-radius: 4px;
      background-color: @fusion-background-color;
    }
    .icon {
      height: 20px;
      width: 25px;
    }
  }
}
