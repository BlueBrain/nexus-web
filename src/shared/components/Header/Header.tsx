import React from 'react';
import { Link } from 'react-router-dom';
import { useLocation } from 'react-router';
import { Menu, Dropdown, MenuItemProps } from 'antd';
import {
  UserOutlined,
  BookOutlined,
  SettingOutlined,
  FileTextOutlined,
  LinkOutlined,
  LogoutOutlined,
  CopyOutlined,
  MenuOutlined,
  PlusOutlined,
} from '@ant-design/icons';
import { useDispatch, useSelector } from 'react-redux';
import { UISettingsActionTypes } from '../../store/actions/ui-settings';
import { RootState } from '../../store/reducers';
import { updateAboutModalVisibility } from '../../store/actions/modals';
import { triggerCopy as copyCmd } from '../../utils/copy';
import { AdvancedModeToggle } from '../../molecules';
import useNotification from '../../hooks/useNotification';
import fusionLogo from '../../images/fusion.svg';

import './Header.scss';

export interface HeaderProps {
  name?: string;
  token?: string;
  environment: string;
  logoImg: string;
  handleLogout: MenuItemProps['onClick'];
  children?: React.ReactChild;
}
const headerIconStyle = { marginRight: 4 };
const Header: React.FunctionComponent<HeaderProps> = ({
  name,
  token,
  environment,
  logoImg,
  handleLogout,
}) => {
  const dispatch = useDispatch();
  const location = useLocation();
  const notification = useNotification();
  const { openCreationPanel } = useSelector(
    (state: RootState) => state.uiSettings
  );
  const notShowDefaultHeader =
    (!token && location.pathname === '/') || location.pathname === '/login';
  const copyTokenCmd = () => {
    if (token) {
      copyCmd(token);
      notification.success({
        message: 'Token copied to clipboard üìÑ',
        position: 'bottomRight',
      });
    }
  };
  const openAboutModal = () => dispatch(updateAboutModalVisibility(true));
  const menu = (
    <Menu
      className="ant-menu-inline"
      items={[
        {
          key: 'header-menu-my-profile',
          label: (
            <Link to="/user">
              <UserOutlined style={headerIconStyle} />
              My Profile
            </Link>
          ),
          className: 'link-menu-item',
        },
        {
          key: 'header-menu-my-date',
          label: (
            <Link to="/my-data">
              <MenuOutlined style={headerIconStyle} />
              My data
            </Link>
          ),
          className: 'link-menu-item',
        },
        ...(token
          ? [
              {
                key: 'header-menu-my-token',
                label: (
                  <>
                    <CopyOutlined style={headerIconStyle} />
                    Copy token
                  </>
                ),
                onClick: () => copyTokenCmd(),
              },
            ]
          : []),
        {
          key: 'header-menu-resources',
          label: (
            <>
              <BookOutlined style={headerIconStyle} />
              <span>Resources</span>
            </>
          ),
          className: 'submenu-overlay-custom',
          popupClassName: 'submenu-overlay-custom-popup',
          children: [
            {
              key: 'header-menu-resources-docs',
              label: (
                <a
                  rel="noopener noreferrer"
                  target="_blank"
                  href="https://bluebrainnexus.io/docs/index.html"
                >
                  <FileTextOutlined style={headerIconStyle} />
                  <span>Documentation</span>
                </a>
              ),
            },
            {
              key: 'header-menu-resources-web',
              label: (
                <a
                  rel="noopener noreferrer"
                  target="_blank"
                  href="https://bbp.epfl.ch/nexus/webprotege/"
                >
                  <LinkOutlined style={headerIconStyle} />
                  <span>Web Prot√©g√©</span>
                </a>
              ),
            },
            {
              key: 'header-menu-resources-atlas',
              label: (
                <a
                  rel="noopener noreferrer"
                  target="_blank"
                  href={
                    environment === 'dev'
                      ? 'http://cell-atlas.kcpdev.bbp.epfl.ch/'
                      : 'https://bbp.epfl.ch/nexus/cell-atlas/'
                  }
                >
                  <LinkOutlined style={headerIconStyle} />
                  <span>Atlas</span>
                </a>
              ),
            },
          ],
        },
        {
          key: 'header-menu-about',
          className: 'link-menu-item',
          onClick: () => openAboutModal(),
          label: (
            <>
              <SettingOutlined style={headerIconStyle} />
              About
            </>
          ),
        },
        {
          key: 'logout',
          className: 'menu-item-logout',
          onClick: event => {
            if (handleLogout) {
              handleLogout(event);
            }
          },
          label: (
            <>
              <LogoutOutlined style={headerIconStyle} />
              Logout
            </>
          ),
        },
      ]}
    />
  );
  const showCreationPanel = location.pathname === '/search';
  const handleOpenCreationPanel = () =>
    dispatch({ type: UISettingsActionTypes.CHANGE_HEADER_CREATION_PANEL });
  if (notShowDefaultHeader) return null;
  return (
    <header id="main-header" className="Header">
      <div className="logo-container">
        <Link to="/">
          <div className="logo-container__logo">
            <img src={logoImg || fusionLogo} alt="Logo" />
          </div>
        </Link>
      </div>
      {token ? (
        <div className="menu-block">
          {name && <AdvancedModeToggle />}
          {name && showCreationPanel && (
            <div
              role="button"
              className="menu-open-creation-panel"
              onClick={handleOpenCreationPanel}
            >
              <PlusOutlined
                rotate={openCreationPanel ? 45 : 90}
                style={{ color: 'white', fontSize: 18 }}
              />
            </div>
          )}
          {name && (
            <Dropdown
              trigger={['click']}
              dropdownRender={() => menu}
              overlayClassName="menu-overlay-custom"
              getPopupContainer={() => document.getElementById('main-header')!}
            >
              <a className="menu-dropdown ant-dropdown-link">
                <UserOutlined />
                <span>{name}</span>
              </a>
            </Dropdown>
          )}
        </div>
      ) : (
        <div className="menu-block">
          <Link to="/login" className="menu-dropdown ant-dropdown-link">
            <UserOutlined />
            <span>Login</span>
          </Link>
        </div>
      )}
    </header>
  );
};

export default Header;
