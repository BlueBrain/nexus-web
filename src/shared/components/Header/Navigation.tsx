import { AppstoreFilled } from '@ant-design/icons';
import { Button, Popover } from 'antd';
import * as React from 'react';
import { Link } from 'react-router-dom';
import './Navigation.less';

const InternalNavigationItem: React.FC<{
  appKey: string;
  appRoute: string;
  onClickNavigationItem: () => void;
  appIcon: string;
  appLabel: string;
}> = ({ appKey, appRoute, onClickNavigationItem, appIcon, appLabel }) => {
  return (
    <NavigationItemContainer appKey={appKey}>
      <Link to={appRoute} title="" onClick={onClickNavigationItem}>
        <img src={appIcon} className="icon" />
        <div className="navigation-text">{appLabel}</div>
      </Link>
    </NavigationItemContainer>
  );
};

const ExternalNavigationItem: React.FC<{
  appKey: string;
  appURL: string;
  onClickNavigationItem: () => void;
  appIcon: string;
  appLabel: string;
}> = ({ appKey, appURL, onClickNavigationItem, appIcon, appLabel }) => {
  return (
    <NavigationItemContainer appKey={appKey}>
      <a
        title=""
        target="_blank"
        href={appURL || ''}
        onClick={onClickNavigationItem}
      >
        <img src={appIcon} className="icon" />
        <div className="navigation-text">
          {appLabel}
          {<>&#x2197;</>}
        </div>
      </a>
    </NavigationItemContainer>
  );
};

const NavigationItemContainer: React.FC<{
  appKey: string;
  children: React.ReactNode;
}> = ({ appKey, children }) => (
  <div key={appKey} className="navigation__item">
    <Button>{children}</Button>
  </div>
);

const Navigation: React.FC<{ subApps: any; authenticated: boolean }> = ({
  subApps,
  authenticated,
}) => {
  const [navMenuVisible, setNavMenuVisible] = React.useState(false);

  return (
    <Popover
      visible={navMenuVisible}
      onVisibleChange={visible => setNavMenuVisible(visible)}
      content={
        <div>
          <div className="navigation">
            {subApps.map((app: any) => {
              return app.subAppType === 'external' ? (
                <ExternalNavigationItem
                  key={app.key}
                  appKey={app.key}
                  appURL={app.url}
                  appIcon={app.icon}
                  appLabel={app.label}
                  onClickNavigationItem={() => setNavMenuVisible(false)}
                />
              ) : app.requireLogin && !authenticated ? (
                <></>
              ) : (
                <InternalNavigationItem
                  key={app.key}
                  appKey={app.key}
                  appRoute={app.route}
                  appIcon={app.icon}
                  appLabel={app.label}
                  onClickNavigationItem={() => setNavMenuVisible(false)}
                />
              );
            })}
          </div>
        </div>
      }
      trigger="click"
      placement="bottomRight"
    >
      <Button
        className="navigation-button"
        style={{ backgroundColor: 'transparent' }}
        icon={<AppstoreFilled style={{ color: '#fff' }} />}
      ></Button>
    </Popover>
  );
};

export default Navigation;
