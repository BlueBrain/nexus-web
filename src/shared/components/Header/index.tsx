import * as React from 'react';
import { Menu, Dropdown, Popover } from 'antd';
import {
  BookOutlined,
  GithubOutlined,
  DownOutlined,
  LoginOutlined,
} from '@ant-design/icons';
import { Realm } from '@bbp/nexus-sdk';
import Copy from '../Copy';
import ConsentPreferences from '../ConsentPreferences';
import { ConsentType } from '../../layouts/FusionMainLayout';

import './Header.less';
import Navigation from './Navigation';

declare var Version: string;

const epflLogo = require('../../images/EPFL-logo.svg');
const infoIcon = require('../../images/infoIcon.svg');
const copyIcon = require('../../images/copyIcon.svg');

const documentationURL = 'https://bluebrainnexus.io/docs';
const repoUrl = 'https://github.com/BlueBrain/nexus-web';
const releaseNoteUrl = 'https://github.com/BlueBrain/nexus-web/releases';

interface InformationContentProps {
  version: string;
  githubIssueURL: string;
  commitHash?: string;
  consent?: ConsentType;
  onClickRemoveConsent?(): void;
}

const InformationContent = (props: InformationContentProps) => {
  return (
    <>
      <p>Nexus is Open Source and available under the Apache 2 License. </p>
      <p>
        Â© 2017-2020
        <a href="https://www.epfl.ch/" target="_blank">
          <img
            style={{ width: '3em', marginBottom: 3 }}
            src={epflLogo}
            alt="EPFL"
          />
        </a>
        {'| '}
        <a href="https://bluebrain.epfl.ch/" target="_blank">
          <span className="bbp-logo">Blue Brain Project</span>
        </a>
      </p>
      <h4>Nexus Services</h4>
      <p>
        Nexus Delta v{props.version} <br />
        <a href={`${repoUrl}/commits/${props.commitHash}`} target="_blank">
          Nexus Fusion {Version}
        </a>
      </p>
      <p>
        <a href={documentationURL} target="_blank">
          <BookOutlined /> Documentation
        </a>
        {' | '}
        <a href={props.githubIssueURL} target="_blank">
          <GithubOutlined /> Report Issue
        </a>
        {' | '}
        <a href={releaseNoteUrl} target="_blank">
          <GithubOutlined /> Fusion Release Notes
        </a>
      </p>
      {
        <ConsentPreferences
          onClickRemove={props.onClickRemoveConsent}
          consent={props.consent}
        />
      }
    </>
  );
};
export interface HeaderProps {
  version: string;
  githubIssueURL: string;
  forgeLink: string;
  name?: string;
  token?: string;
  links?: React.ReactNode[];
  realms: Realm[];
  displayLogin?: boolean;
  children?: React.ReactChild;
  consent?: ConsentType;
  commitHash?: string;
  dataCart?: React.ReactNode;
  onClickRemoveConsent?(): void;
  performLogin(realmName: string): void;
  subApps: any;
  authenticated: boolean;
}

const Header: React.FunctionComponent<HeaderProps> = ({
  name,
  realms,
  token,
  displayLogin = true,
  links = [],
  children,
  version,
  githubIssueURL,
  forgeLink,
  consent,
  commitHash,
  dataCart,
  onClickRemoveConsent,
  performLogin,
  subApps,
  authenticated,
}) => {
  const menu = (
    <Menu>
      {links.map((link, i) => (
        <Menu.Item key={i}>{link}</Menu.Item>
      ))}
    </Menu>
  );

  const realmsFilter = realms.filter(
    r => r._label !== 'serviceaccounts' && !r._deprecated
  );

  const realmMenu = (
    <Menu>
      {realmsFilter.map((r: Realm, i: number) => (
        <Menu.Item
          key={i}
          title={r.name}
          onClick={e => {
            e.domEvent.preventDefault();
            performLogin(realmsFilter[i].name);
          }}
        >
          {r.name}
        </Menu.Item>
      ))}
    </Menu>
  );

  return (
    <header className="Header">
      <div className="selectors">{children}</div>
      <div className="menu-block">
        {name && forgeLink !== '' && (
          <a href={forgeLink} target="_blank" className="forge-button">
            Forge Templates
          </a>
        )}
        {token && (
          <Copy
            textToCopy={token}
            render={(copySuccess, triggerCopy) => (
              <button
                className="copy-token-button"
                onClick={() => {
                  triggerCopy();
                }}
              >
                <img src={copyIcon} />{' '}
                {copySuccess ? (
                  <span className="button-text">Copied!</span>
                ) : (
                  <span className="button-text">Copy token</span>
                )}
              </button>
            )}
          />
        )}
        {dataCart}
        <Popover
          content={
            <InformationContent
              version={version}
              githubIssueURL={githubIssueURL}
              consent={consent}
              onClickRemoveConsent={onClickRemoveConsent}
              commitHash={commitHash}
            />
          }
          trigger="click"
          title="Information"
          placement="bottomRight"
        >
          <img
            src={infoIcon}
            className="ui-header-info-button"
            alt="Information"
          />
        </Popover>
        <Navigation authenticated={authenticated} subApps={subApps} />
        {name ? (
          <Dropdown overlay={menu}>
            <a className="menu-dropdown ant-dropdown-link">
              {name} <DownOutlined />
            </a>
          </Dropdown>
        ) : displayLogin ? (
          <Dropdown overlay={realmMenu}>
            <a className="menu-dropdown ant-dropdown-link">
              login <LoginOutlined />
            </a>
          </Dropdown>
        ) : null}
      </div>
    </header>
  );
};

export default Header;
