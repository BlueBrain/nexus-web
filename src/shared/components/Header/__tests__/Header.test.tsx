import * as React from 'react';
import { shallow, mount } from 'enzyme';
import Header from '..';
import { MemoryRouter } from 'react-router-dom';

jest.mock('react-redux', () => {
  const ActualReactRedux = require.requireActual('react-redux');
  return {
    ...ActualReactRedux,
    useSelector: jest.fn().mockImplementation(() => {
      return 'mockState';
    }),
  };
});

const links: React.ReactNode[] = [
  <p>A p tag</p>,
  <a href="/somepage">A link to some page</a>,
];

const shallowHeader = shallow(
  <Header
    name="Mark Hamill"
    realms={[]}
    serviceAccountsRealm="serviceaccounts"
    links={links}
    githubIssueURL=""
    forgeLink=""
    version=""
    performLogin={(name: string) => {}}
    subApps={[]}
    authenticated={false}
  />
);

const wrapper = mount(
  <MemoryRouter>
    <Header
      name="Mark Hamill"
      realms={[]}
      serviceAccountsRealm="serviceaccounts"
      links={links}
      githubIssueURL=""
      forgeLink=""
      version=""
      performLogin={(name: string) => {}}
      subApps={[]}
      authenticated={false}
    />
  </MemoryRouter>
);

describe('Header component', () => {
  it('Should render correctly', () => {
    expect(shallowHeader).toMatchSnapshot();
  });

  it('Should have a logo block and a menu block', () => {
    expect(shallowHeader.find('.logo-block')).toBeTruthy();
    expect(shallowHeader.find('.menu-block')).toBeTruthy();
  });

  describe('Menu Block', () => {
    it('Should display the user name followed by a space', () => {
      expect(wrapper.find('.menu-dropdown').text()).toEqual('Mark Hamill ');
    });

    it('Should display login link', () => {
      const wrapperNoName = mount(
        <MemoryRouter>
          <Header
            name=""
            realms={[]}
            serviceAccountsRealm="serviceaccounts"
            links={links}
            githubIssueURL=""
            forgeLink=""
            version=""
            performLogin={(name: string) => {}}
            subApps={[]}
            authenticated={false}
          />
        </MemoryRouter>
      );

      expect(wrapperNoName.find('.menu-dropdown').text()).toEqual('login ');
    });

    it('Should NOT display login link', () => {
      const wrapperNoLogin = mount(
        <MemoryRouter>
          <Header
            name=""
            realms={[]}
            serviceAccountsRealm="serviceaccounts"
            links={links}
            githubIssueURL=""
            forgeLink=""
            version=""
            displayLogin={false}
            performLogin={(name: string) => {}}
            subApps={[]}
            authenticated={false}
          />
        </MemoryRouter>
      );

      expect(wrapperNoLogin.find('.menu-dropdown').children()).toHaveLength(0);
    });
  });
});
