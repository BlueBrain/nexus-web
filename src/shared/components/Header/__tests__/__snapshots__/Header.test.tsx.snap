// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Should render correctly 1`] = `
<header
  className="Header"
>
  <div
    className="selectors"
  />
  <div
    className="menu-block"
  >
    <Popover
      content={
        <InformationContent
          githubIssueURL=""
          version=""
        />
      }
      mouseEnterDelay={0.1}
      mouseLeaveDelay={0.1}
      overlayStyle={Object {}}
      placement="bottomRight"
      title="Information"
      trigger="click"
    >
      <img
        alt="Information"
        className="ui-header-info-button"
        src="/mock/path"
      />
    </Popover>
    <Navigation
      authenticated={false}
      subApps={Array []}
    />
    <Dropdown
      mouseEnterDelay={0.15}
      mouseLeaveDelay={0.1}
      overlay={
        <Menu>
          <MenuItem>
            <p>
              A p tag
            </p>
          </MenuItem>
          <MenuItem>
            <a
              href="/somepage"
            >
              A link to some page
            </a>
          </MenuItem>
        </Menu>
      }
    >
      <a
        className="menu-dropdown ant-dropdown-link"
      >
        Mark Hamill
         
        <ForwardRef(DownOutlined) />
      </a>
    </Dropdown>
  </div>
</header>
`;
